app.coffee

```diff
@@ -54,14 +55,40 @@ app.post "/webhook", (req, res) ->
   # Send a blank response, so slack knows we got it.
   res.send ""
 
+# Ban the nick
+app.post "/user/nick/ban", (req, res)->
+  throw "Invalid Token" unless req.body.token == process.env.OUTGOING_TOKEN_BOT
+
+  if user.ban_nick req.body.nick
+    res.send "ok"
+    return
+
+  res.send "error!"
+
+# Ban the session
+app.post "/user/session/ban", (req, res)->
+  throw "Invalid Token" unless req.body.token == process.env.OUTGOING_TOKEN_BOT
+
+  if user.ban_session req.body.nick
+    res.send "ok"
+    return
+
+  res.send "error!"
```

Comment:
jinora already receives all messages that we type in the public channel. Why add needless complexity by bringing in another dependency that not every might have installed.